src/python/pants/backend/go/subsystems/gotest.py

```diff
@@ -24,3 +28,34 @@ class GoTestSubsystem(Subsystem):
         ),
         passthrough=True,
     )
+
+    coverage_mode = EnumOption(
+        "--cover-mode",
+        default=GoCoverMode.SET,
+        help=softwrap(
+            """\
+            Coverage mode when running Go tests. Valid options are `set`, `count`, and `atomic`.
+            """
+        ),
+    )
+
+    _coverage_output_dir = StrOption(
+        default=str(PurePath("{distdir}", "coverage", "go", "{import_path_escaped}")),
```

Comment:
Hm... I'm surprised that you need this prefix... I guess that the `test` goal doesn't add it?

IIRC, @benjyw refactored the report output location to have a default, which would probably be reasonable to do here too if it makes things easier for you.

Reply:
The `test` goal does not influence the output location. I based this code on the option code in the Python backend which supplies the path prefix itself. See https://github.com/pantsbuild/pants/blob/28d756e679dfe19aa788d86b39500dfef911c388/src/python/pants/backend/python/goals/coverage_py.py#L150-L154